lineman/spec-e2e/group_spec.coffee

```diff
@@ -6,6 +6,35 @@ describe 'Group Page', ->
   discussionForm = require './helpers/discussion_form_helper.coffee'
   threadPage = require './helpers/thread_helper.coffee'
 
+  ddescribe 'group privacy settings', ->
+    describe 'viewing each group privacy type as a non member', ->
+      it 'secret group', ->
+        groupsHelper.loadPath('view_secret_group_as_non_member')
+        groupsHelper.expectCSS('.error-page')
+
+      it 'closed group', ->
+        groupsHelper.loadPath('view_closed_group_as_non_member')
+        groupsHelper.expectCSS('.join-group-button__ask-to-join-group')
+
+      it 'open group', ->
+        groupsHelper.loadPath('view_open_group_as_non_member')
+        groupsHelper.expectCSS('.join-group-button__join-group')
+
+    describe 'allow public discussions in closed group', ->
+      beforeEach ->
+        groupsHelper.loadPath('setup_closed_group')
```

Comment:
I've taken a sudden liking for using generic helper methods like loadPath, click(many, things) and assertCSS rather than lots of custom helper methods.